### ISSUE TITLE
Capture cursor

### ISSUE THREAD

**Issue by geist22** [swfans/syndwarsfx#25]
The cursor should probably be prevented from leaving the window while it is active.

**Comment by mefistotelis** [comment]
I don't want to create too much options for that, but it would make total sense to capture the mouse in relative position mode.
In that mode, the game always keeps "real" mouse near its window center, and warps it to center if it moves too far (in SW, that is 50 pixels). So the game computes mouse movement internally, based on how the user shifted the mouse from center.

In Dungeon Keeper, relative position mode is required for Possession. So I guess it is also used in Magic Carpet.
Since in that mode the cursor is prevented from leaving game window anyway, it makes sense to merge these two. And there could be a command line option in swars to enable it.

**Comment by geist22** [comment]
I imagined the cursor to be captured by default whenever the window is active.
Though a key combination for manual control can work too, with a config or command line option to have it enabled by default.

DOSBox Staging for example doesn't capture the cursor at startup by default but has a config option to do so. Clicking inside the window automatically captures, and you can use a key combination (CTRL+F10) or the middle mouse button to activate/deactivate it manually. It works pretty well, though it would prevent binding the middle mouse button to ingame actions.

**Comment by mefistotelis** [comment]
Related PR:
https://github.com/swfans/syndwarsfx/pull/434

**Comment by geist22** [comment]
Is there something against it being used in windowed mode as well?

You basically can't scroll with the mouse and have a hard time clicking on the super shield bar because the cursor doesn't stop at the border and just exits the window.

**Comment by mefistotelis** [comment]
Yeah, I still think it should be a separate config option, which works regardless of screen mode.

EDIT:
Described my idea of how this should work within the PR:
> Binding the capture mouse function to fullscreen is not really a preferred solution.
>
> What we need is
```
TbResult LbMouseChangeCapture(TbBool enable);
```
> within `bfmouse.h`.
>
> That function should work even without a window, so the capture state needs to be stored within global variable `TbBool lbMouseCapture;`. If a window exists (ie. lbScreenInitialised is true), SDL_SetWindowGrab should also be called within that function (not sure if directly or by a call to update function within screen module.. whatever looks better).
>
> Similarly, the function should be called during window creation, if value of `lbMouseCapture` requests it.

**Comment by geist22** [comment]
Thanks to [brunosaboia](https://github.com/brunosaboia) this has been implemented now 👍

`MouseCapture` option in `CONFIG.INI` accepts `Never`, `Always` or `FullScreen` (default).
